initfiles/componentfiles/configxml/sparkThor.xsl.in

```diff
@@ -38,7 +38,7 @@
         </xsl:call-template>
         <xsl:call-template name="printVariable">
             <xsl:with-param name="var" select="'SPARK_MASTER_HOST'"/>
-            <xsl:with-param name="val" select="Instance/@netAddress"/>
+            <xsl:with-param name="val" select="SparkMasterInstance/@netAddress"/>
```

Comment:
if it's possible to have multiple sparkMasterinstances, we might need to take the first one always.
something like select="SparkMasterInstance[1]/@netAddress"/ ?

Reply:
We should only ever have one master I think.  Perhaps I should try to make that a unique element?

Reply:
Yes, this should be a single element